pkg/image/api/helper_test.go

```diff
@@ -96,14 +107,36 @@ func TestParseDockerImageReference(t *testing.T) {
 			ID:        "sha256:3c87c572822935df60f0f5d3665bd376841a7fcfeb806b5f212de6a00e9a7b25",
 		},
 		{
-			From:      "myregistry.io/foo",
-			Registry:  "myregistry.io",
+			From:     "myregistry.io/foo",
+			Registry: "myregistry.io",
+			Name:     "foo",
+		},
+		{
+			From:     "localhost/bar",
+			Registry: "localhost",
+			Name:     "bar",
+		},
+		{
+			From:      "docker.io/library/myapp",
+			Registry:  "docker.io",
+			Namespace: "library",
+			Name:      "myapp",
+		},
+		{
+			From:      "docker.io/myapp",
+			Registry:  "docker.io",
 			Namespace: DockerDefaultNamespace,
-			Name:      "foo",
+			Name:      "myapp",
 		},
 		{
-			From:      "localhost/bar",
```

Comment:
It didn't get removed. It may seems like it in *unified* view. See the [split view](https://github.com/openshift/origin/pull/6777/files?diff=split#diff-c7ce025f36a3834873271fcff866fdb4R118). Only the namespace part has been removed.